### PR TITLE
Add environment variables documentation

### PR DESCRIPTION
**Pull request by stretch07**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- What does this PR change? Give us a brief description. 
Completes the Stub environment variables page located [here](https://v2.tauri.app/reference/environment-variables/)

### CODE REVIEW

**Comment by stretch07** [comment]
darn, I have no clue why the build is failing, it seems like something more should be escaped in the markdown but i cant figure out what it is

**Comment by vasfvitor** [comment]
hey thanks very much for this, very much needed!

is it possible that you could provide the source for these env var? Or are they a compilation found in multiple places?

I'll take a quick look as why it's failing to build

**Review by vasfvitor — CHANGES_REQUESTED** [review]
Could you change the formatting to this pattern or something that is easier to navigate?

`ENV_VAR` : description
`ENV_VAR` : description
`ENV_VAR` : description

For instance, this page https://deploy-preview-2200--tauri-v2.netlify.app/start/migrate/from-tauri-1/#environment-variables-changes lists as this:

- `ENV_VAR` : description
- `ENV_VAR` : description
- `ENV_VAR` : description

I'm open to ideas, if you have anything in mind...

Some visual references: 
https://vitejs.dev/guide/env-and-mode#env-variables
https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime
https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

**Comment by stretch07** [comment]
> hey thanks very much for this, very much needed!
> 
> is it possible that you could provide the source for these env var? Or are they a compilation found in multiple places?
> 
> I'll take a quick look as why it's failing to build

they're practically directly copied from here:
https://github.com/tauri-apps/tauri/blob/dev/tooling/cli/ENVIRONMENT_VARIABLES.md

I'm not aware of v2 adding more environment variables but if so let me know

**Comment by vasfvitor** [comment]
> they're practically directly copied from here

so for now I think this is good, but for the future a improvement could be made to import this file contents into the documentation as a component, so we get the benefit of having always up to date content. Eventually could add extra information, if needed.

If you're feeling you want to tackle this, let me know, if not, I'll just check the preview and merge this.

**Comment by vasfvitor** [comment]
but now looking at it, the rendered content will be different, have to figure a way to place this tip card nicely. I'll merge this alright

**Comment by vasfvitor** [comment]
thanks again

File: src/content/docs/reference/environment-variables.mdx [hunk 1]
```diff
@@ -2,15 +2,56 @@
...
-    variant: caution
 ---
 
-import Stub from '@components/Stub.astro';
+This is a documentation of all environment variables used by tauri core crates and tauri CLI.
 
-<Stub>
+## Tauri CLI
 
-Integrate https://github.com/tauri-apps/tauri/blob/dev/tooling/cli/ENVIRONMENT_VARIABLES.md into this page.
```
Comment (vasfvitor) [inline src/content/docs/reference/environment-variables.mdx:14]: -_-

just saw this lol

**Comment by stretch07** [comment]
> > they're practically directly copied from here
> 
> so for now I think this is good, but for the future a improvement could be made to import this file contents into the documentation as a component, so we get the benefit of having always up to date content. Eventually could add extra information, if needed.
> 
> If you're feeling you want to tackle this, let me know, if not, I'll just check the preview and merge this.

that's actually a great idea, I'll get familiar with astro and look into how to do this. thanks!

**Comment by vasfvitor** [comment]
we have a few pages that does this, here you can see one that is in progress but mostly done, https://github.com/tauri-apps/tauri-docs/blob/137bb35ed873efb0536368ba751ae085a60ad12f/src/components/Permissions.astro, just need to fix something upstream.

other than that you can take a look at the dir /packages that has [config-generator](https://github.com/tauri-apps/tauri-docs/tree/v2/packages/config-generator), [js-api-generator](https://github.com/tauri-apps/tauri-docs/tree/v2/packages/js-api-generator), [releases-generator](https://github.com/tauri-apps/tauri-docs/tree/v2/packages/releases-generator) that mix data importing and handling

overall everything that can be done this way, should be.

**Pull request merged**